src/modelfree/scheduling.py

```diff
@@ -1,31 +1,55 @@
 from abc import ABC, abstractmethod
+import collections
 import functools
+import itertools
+import operator
 
 
 class Scheduler(object):
     """Keep track of frac_remaining and return time-dependent values"""
-    def __init__(self, func_dict=None):
-        if func_dict is None:
-            func_dict = {}
-        assert isinstance(func_dict, dict)
-        self.func_dict = func_dict
-        self.allowed_func_types = ['lr', 'rew_shape', 'noise']
+    def __init__(self, annealer_dict=None):
+        if annealer_dict is None:
+            annealer_dict = {}
+        self.annealer_dict = annealer_dict
+        self.func_dict = {}
+        for k, v in annealer_dict.items():
+            self.func_dict[k] = v.get_value
+
+        self.conditionals = collections.defaultdict(lambda: False)
         self.frac_remaining = 1  # frac_remaining goes from 1 to 0
 
+    @staticmethod
+    def _validate_func_type(func_type):
+        allowed_func_types = frozenset(['lr', 'rew_shape', 'noise'])
+        if func_type not in allowed_func_types:
+            raise KeyError("func_type not in allowed_func_types")
+
     def _update_frac_remaining(self, frac_remaining=None):
         if frac_remaining is not None:
             self.frac_remaining = frac_remaining
 
-    def set_func(self, func_type, func):
+    def set_conditional(self, func_type):
+        """Set the value of key func_type to be True in the dictionary self.conditionals"""
```

Comment:
Docstrings should describe the interface (from the users perspective) first. Implementation details can be worth mentioning but are best not the focus. Why would I want to call this function, what does `self.conditionals` do?

Reply:
Instructive comment. Thanks.